python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -480,6 +483,91 @@ def close(self) -> None:
         self._worker.core_worker.experimental_channel_set_error(self._reader_ref)
 
 
+@DeveloperAPI
+class BufferedSharedMemoryChannel(ChannelInterface):
+    """A Channel that allows to buffer inputs.
+
+    If the buffer is full for writes or empty for reads and the operation
+    blocks longer than configured timeouts, RayChannelTimeoutError will be
+    raised.
+
+    As long as the buffer is not full, read/write APIs are non-blocking.
+    The caller of the API should guarantee to consume buffers before they are
+    exhausted.
```

Comment:
let's be more accurate by talking about `read` and `write` which are the APIs. It is vague what "consume buffers" mean.

Also, we can just talk about a timeout exception will be thrown if execution blocks.

Reply:
made it more concrete